implementation/src/main/java/io/smallrye/config/SecretKeysConfigSourceInterceptor.java

```diff
@@ -12,15 +14,15 @@
 public class SecretKeysConfigSourceInterceptor implements ConfigSourceInterceptor {
     private static final long serialVersionUID = 7291982039729980590L;
 
-    private final Set<String> secrets;
+    private final Set<PropertyName> secrets;
 
     public SecretKeysConfigSourceInterceptor(final Set<String> secrets) {
-        this.secrets = secrets;
+        this.secrets = secrets.stream().map(PropertyName::new).collect(toSet());
     }
 
     @Override
     public ConfigValue getValue(final ConfigSourceInterceptorContext context, final String name) {
-        if (SecretKeys.isLocked() && secrets.contains(name)) {
+        if (SecretKeys.isLocked() && secrets.contains(new PropertyName(name))) {
```

Comment:
Creating a new `PropertyTime` every time `getValue` is called may impact performance, perhaps consider iterating the `Set` and comparing the name instead?

Reply:
It was to move forward with the implementation. I haven't profiled the code yet, but I am not too worried about it because when we are populating the config, the secrets are unlocked, but I'll have a look. Thanks!